lib/Onyx.ts

```diff
@@ -632,42 +634,38 @@ function update(data: OnyxUpdate[]): Promise<void> {
         }
     };
 
+    const handleCollectionOperation = (operation: typeof enqueueSetOperation | typeof enqueueMergeOperation, onyxMethod: OnyxMethod) => {
+        return (key: OnyxKey, value: OnyxInput<OnyxKey>) => {
+            const collection = value as Collection<CollectionKey, unknown, unknown>;
+            if (!OnyxUtils.isValidNonEmptyCollectionForMerge(collection)) {
+                Logger.logInfo(`${onyxMethod} enqueued within update() with invalid or empty value. Skipping this operation.`);
+                return;
+            }
+
+            const collectionKeys = Object.keys(collection);
+            if (OnyxUtils.doAllCollectionItemsBelongToSameParent(key, collectionKeys)) {
+                const mergedCollection: OnyxInputKeyValueMapping = collection;
+                collectionKeys.forEach((collectionKey) => operation(collectionKey, mergedCollection[collectionKey]));
+            }
+        };
+    };
+
     const promises: Array<() => Promise<void>> = [];
     let clearPromise: Promise<void> = Promise.resolve();
 
     data.forEach(({onyxMethod, key, value}) => {
-        switch (onyxMethod) {
-            case OnyxUtils.METHOD.SET:
-                enqueueSetOperation(key, value);
-                break;
-            case OnyxUtils.METHOD.MERGE:
-                enqueueMergeOperation(key, value);
-                break;
-            case OnyxUtils.METHOD.MERGE_COLLECTION: {
-                const collection = value as Collection<CollectionKey, unknown, unknown>;
-                if (!OnyxUtils.isValidNonEmptyCollectionForMerge(collection)) {
-                    Logger.logInfo('mergeCollection enqueued within update() with invalid or empty value. Skipping this operation.');
-                    break;
-                }
-
-                // Confirm all the collection keys belong to the same parent
-                const collectionKeys = Object.keys(collection);
-                if (OnyxUtils.doAllCollectionItemsBelongToSameParent(key, collectionKeys)) {
-                    const mergedCollection: OnyxInputKeyValueMapping = collection;
-                    collectionKeys.forEach((collectionKey) => enqueueMergeOperation(collectionKey, mergedCollection[collectionKey]));
-                }
-
-                break;
-            }
-            case OnyxUtils.METHOD.MULTI_SET:
-                Object.entries(value).forEach(([entryKey, entryValue]) => enqueueSetOperation(entryKey, entryValue));
-                break;
-            case OnyxUtils.METHOD.CLEAR:
+        const handlers: Record<OnyxMethodMap[keyof OnyxMethodMap], (k: typeof key, v: typeof value) => void> = {
```

Comment:
@arosiclair This PR fixes the bug where Onyx.update() doesn't work with `setCollection`. Initially we had [another PR](https://github.com/Expensify/react-native-onyx/pull/594) to add the new `setCollection` method, but everyone missed the part that this new `setCollection` method needs to also work when using Onyx.update().

> Issue: `setCollection` is not working when invoked via `Onyx.update()` , [more details here ](https://callstack-hq.slack.com/archives/C05LX9D6E07/p1732212537404029?thread_ts=1724780554.260709&cid=C05LX9D6E07)

As discussed in the above thread (attached to this PR's description), we need to improve the development experience to not miss these type of bugs in the feature, i.e. the TS both locally and on CI should flag that we are missing to handle any newly added methods. Having an object mapping allows us to have more strict type checks and to prevent these type of issues in the future.

Reply:
Oh okay I see now. In that case, can we also add a unit test to cover `setCollection` with `Onyx.update`?

Reply:
@arosiclair I've added tests

Reply:
Thanks!